### ISSUE TITLE
Crash when attaching a text to the description of an existing card

### ISSUE THREAD

**Issue by reox** [stefan-niedermann/nextcloud-deck#1344]
Please use **[GitHub reactions](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/)** 👍 to show that you are affected by the same issue. Please don't comment if you have no relevant information to add!

**Describe the bug**

When I try to share a link, the deck app crashes.

1. open a website in firefox/chrome (tested both)
2. share the link to the page (alternative: also sharing any text provokes the same crash)
3. click on 'add to existing card'
4. select a card
5. select "add to description" (adding as comment seems to work fine)
6. crash

The text seems to be added fine however.

**Expected behavior**
The link should be attached to the description

**Versions**

 - Nextcloud: 24.0.5
 - Nextcloud Deck: 1.7.1
 - Nextcloud Android: 3.21.2
 - Nextcloud Android Deck: 1.20.2

**Smartphone (please complete the following information):**
- Device: 
- Android-Version: 
- App-Store:
  - [x] Google Play Store
  - [ ] Google Play Store (Beta channel)
  - [ ] F-Droid
  - [ ] Huawei AppGallery

**Stacktrace**

```
App Version: 1.20.2
App Version Code: 1020002
Server App Version: 1.7.1
App Flavor: play

Files App Version Code: 30210290

---

OS Version: 4.9.118-23733677(G960FXXUHFVG4)
OS API Level: 29
Device: starlte
Manufacturer: samsung
Model (and Product): SM-G960F (starltexx)

---

java.lang.RuntimeException: Can't toast on a thread that has not called Looper.prepare()
	at android.widget.Toast$TN.<init>(Toast.java:949)
	at android.widget.Toast.<init>(Toast.java:208)
	at android.widget.Toast.makeText(Toast.java:617)
	at android.widget.Toast.makeText(Toast.java:586)
	at it.niedermann.nextcloud.deck.ui.sharetarget.ShareTargetActivity$2.onResponse(ShareTargetActivity.java:167)
	at it.niedermann.nextcloud.deck.ui.sharetarget.ShareTargetActivity$2.onResponse(ShareTargetActivity.java:164)
	at it.niedermann.nextcloud.deck.persistence.sync.SyncManager$13.onResponse(SyncManager.java:1216)
	at it.niedermann.nextcloud.deck.persistence.sync.SyncManager$13.onResponse(SyncManager.java:1213)
	at it.niedermann.nextcloud.deck.persistence.sync.helpers.providers.CardDataProvider.goDeeperForUpSync(CardDataProvider.java:362)
	at it.niedermann.nextcloud.deck.persistence.sync.helpers.SyncHelper$3.onResponse(SyncHelper.java:161)
	at it.niedermann.nextcloud.deck.persistence.sync.helpers.SyncHelper$3.onResponse(SyncHelper.java:153)
	at it.niedermann.nextcloud.deck.api.RequestHelper$ResponseConsumer.accept(RequestHelper.java:52)
	at io.reactivex.internal.observers.LambdaObserver.onNext(LambdaObserver.java:63)
	at io.reactivex.internal.operators.observable.ObservableSubscribeOn$SubscribeOnObserver.onNext(ObservableSubscribeOn.java:58)
	at io.reactivex.internal.operators.observable.ObservableMap$MapObserver.onNext(ObservableMap.java:62)
	at io.reactivex.internal.operators.observable.ObservableFromPublisher$PublisherSubscriber.onNext(ObservableFromPublisher.java:56)
	at com.nextcloud.android.sso.api.NextcloudAPI.lambda$performRequestObservableV2$1$com-nextcloud-android-sso-api-NextcloudAPI(NextcloudAPI.java:130)
	at com.nextcloud.android.sso.api.NextcloudAPI$$ExternalSyntheticLambda2.subscribe(Unknown Source:6)
	at io.reactivex.internal.operators.observable.ObservableFromPublisher.subscribeActual(ObservableFromPublisher.java:31)
	at io.reactivex.Observable.subscribe(Observable.java:12284)
	at io.reactivex.internal.operators.observable.ObservableMap.subscribeActual(ObservableMap.java:32)
	at io.reactivex.Observable.subscribe(Observable.java:12284)
	at io.reactivex.internal.operators.observable.ObservableSubscribeOn$SubscribeTask.run(ObservableSubscribeOn.java:96)
	at io.reactivex.internal.schedulers.ScheduledDirectTask.call(ScheduledDirectTask.java:38)
	at io.reactivex.internal.schedulers.ScheduledDirectTask.call(ScheduledDirectTask.java:26)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:919)

```

**Comment by stefan-niedermann** [comment]
Thanks for the report, we'll have a look at it, but i guess a fix should be quite easy 🙂

**Comment by stefan-niedermann** [comment]
Will be fixed in version `1.20.3`
